tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.cpp

```diff
@@ -341,14 +347,19 @@ namespace tt {
 
 namespace tt_metal {
 
-inline void validate_supported_arch(tt::ARCH arch, UnaryOpType op_type) {
+inline void validate_supported_arch(tt::ARCH arch, DataType input_datatype, DataType output_datatype, UnaryOpType op_type) {
     switch (op_type) {
         case UnaryOpType::REMAINDER:
         case UnaryOpType::FLOOR:
         case UnaryOpType::LEFT_SHIFT:
         case UnaryOpType::RIGHT_SHIFT:
             TT_FATAL(arch == tt::ARCH::WORMHOLE_B0, "Op is only supported on Wormhole");
             break;
+        case UnaryOpType::FMOD:
+            TT_FATAL(arch == tt::ARCH::WORMHOLE_B0, "Op is only supported on Wormhole");
+            TT_FATAL(input_datatype == DataType::BFLOAT16, "Data type is not supported for Bitwise operations");
+            TT_FATAL(output_datatype == DataType::BFLOAT16, "Data type is not supported for Bitwise operations");
```

Comment:
Is there a reason it is specifically bfloat16, or can it be any floating point type. If can be any floating point type, you can use `is_floating_point(dtype)`. The description is also incorrect (not a bitwise op)

Reply:
Hi Austin, the results match for bfloat16. There is a mismatch in TT result when other fp data types are used.